packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { I18n } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type LanguageGroup = {
+	id: string;
+	name: ( translate: I18n[ '__' ] ) => string;
+	subTerritories?: string[];
+	countries?: string[];
+	default?: boolean;
+};
+
+type LanguageSlug = string;
+type WPLocale = string;
+
+type BaseLanguage = {
+	langSlug: LanguageSlug;
+	name: string;
+	popular?: number;
+	rtl?: boolean;
+	territories: string[];
+	value: number;
+	wpLocale: WPLocale | '';
+};
+
+type SubLanguage = BaseLanguage & { parentLangSlug: string };
+
+export type Language = BaseLanguage | SubLanguage;
+
+type Props = {
+	onSelectLanguage: ( language: Language ) => void;
+	languages: Language[];
+	languageGroups: LanguageGroup[];
+	defaultLananguageGroupId: string;
+};
+
+const LanguagePicker = ( {
+	onSelectLanguage,
+	languages,
+	languageGroups,
+	defaultLananguageGroupId,
+}: Props ) => {
+	const { __ } = useI18n();
+	const [ filter, setFilter ] = useState( defaultLananguageGroupId );
+
+	const getFilteredLanguages = () => {
+		switch ( filter ) {
+			case 'popular':
+				return languages
+					.filter( ( language ) => language.popular )
+					.sort( ( a, b ) => ( a.popular as number ) - ( b.popular as number ) );
+			default: {
+				const languageGroup = languageGroups.find( ( l ) => l.id === filter );
+				const subTerritories = languageGroup ? languageGroup.subTerritories : [];
+				return languages
+					.filter( ( language ) =>
+						language.territories.some( ( t ) => subTerritories?.includes( t ) )
+					)
+					.sort( ( a, b ) => a.name.localeCompare( b.name ) );
+			}
+		}
+	};
+
+	const renderCategoryButtons = () => {
+		return languageGroups.map( ( languageGroup ) => {
+			const isSelected = filter === languageGroup.id;
+
+			const onClick = () => {
+				setFilter( languageGroup.id );
+			};
+
+			return (
+				<div key={ languageGroup.id }>
+					<Button onClick={ onClick } className="language-picker__language-group">
+						<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name( __ ) }</span>
+					</Button>
+				</div>
+			);
+		} );
+	};
+
+	return (
+		<>
+			<div className="language-picker__regions-label">{ __( 'regions' ) }</div>
```

Comment:
I see we're using `@wordpress/i18n` here. This relies on the locale data to be already set. AFAIK, `CalypsoI18nProvider` currently does that for both Calypso-style localizations and Gutenberg-style localizations. But using that in a package assumes that this is somehow handled already in the consumer of the package. What are your thoughts about this?

also cc @automattic/i18n about it

Reply:
If there's no provider, `@automattic/react-i18n` will use `@worpdress/i18n` instance with empty locale data, and therefore string won't get translated. A good proposal for `@automattic/react-i18n` might be to fall back to the default `@wordpress/i18n` instead of creating an empty one.

Reply:
Indeed, I've also explained more about that problem in https://github.com/Automattic/wp-calypso/pull/46020#discussion_r504487702

Reply:
Thanks again!

I'll update the README as you suggested.

One thing though, the package no longer uses `react-i18n`, only `wordpress/i18n`. Do you feel that it _should_ be using `react-i18n`? (Admittedly I only changed it because using `react-i18n` was breaking the storybook in a mysterious way)

Reply:
I wouldn't say it's a hard requirement as to which one to use. `react-i18n` is handy, because it adds some syntactic features on top of `@wordpress/i18n`, but since it is generally a wrapper around `@wordpress/i18n`, it doesn't really matter. It basically boils down to:

* Use `react-i18n` if you need to use the HoC and/or the React hook.
* Use `@wordpress/i18n` if you don't need to use any of the above.

Theoretically, it can turn out to be a blocker if `react-i18n` uses an older or newer version of `@wordpress/i18n`, but this shouldn't be a difficult problem to solve, as we have full control over `reach-i18n`, especially if it happens in Calypso.